R/toolbar.R

```diff
@@ -26,3 +26,57 @@ toolbar <- function(
   tag_require(tag, version = 5, caller = "toolbar()")
   as_fragment(tag)
 }
+
+#' Toolbar Input Select
+#'
+#' @description
+#' Create a select list input control that can be used to choose a single item
+#' from a list of values, suitable for use within a [toolbar()]. This is a
+#' wrapper around [shiny::selectInput()] with `selectize = FALSE` and
+#' appropriate styling for toolbar usage.
+#'
+#' @examplesIf rlang::is_interactive()
+#' toolbar(
+#'   align = "right",
+#'   toolbar_input_select(
+#'     id = "select",
+#'     choices = c("Option 1", "Option 2", "Option 3"),
+#'     selected = "Option 2"
+#'   )
+#' )
+#'
+#' @param id The input ID.
+#' @param choices List of values to select from. If elements of the list are
+#'   named, then that name - rather than the value - is displayed to the user.
+#'   It's also possible to group related inputs by providing a named list whose
+#'   elements are (either named or unnamed) lists, vectors, or factors. In this
+#'   case, the outermost names will be used as the group labels (leveraging the
+#'   `<optgroup>` HTML tag) for the elements in the respective sublist.
+#' @param selected The initially selected value. If not specified then defaults
+#'   to the first value.
+#' @param width The width of the input, e.g. '400px', or '100%'; see
+#'   [validateCssUnit()].
+#'
+#' @return Returns a select input control suitable for use in a toolbar.
+#'
+#' @family Toolbar components
+#' @export
+toolbar_input_select <- function(
+  id,
+  choices,
+  selected = NULL,
+  width = NULL
+) {
+  htmltools::div(
+    class = "bslib-toolbar-input-select form-select-sm",
+    shiny::selectInput(
+      id,
+      label = NULL, # Removed for slimline input component
+      choices = choices,
+      selected = selected,
+      multiple = FALSE,
+      selectize = FALSE,
+      width = width
```

Comment:
`size = NULL` is the default for `selectInput()` and usually I wouldn't include it here, but if it's set you get a very different select input so it's worth setting this to ensure it doesn't accidentally change on us later.

```suggestion
      width = width,
      size = NULL
```